subprojects/core/src/main/java/org/gradle/api/internal/tasks/DefaultTaskInputs.java

```diff
@@ -146,9 +146,17 @@ public FileCollection getSourceFiles() {
         return allSourceFiles;
     }
 
+    @Override
     public Map<String, Object> getProperties() {
+        DeprecationLogger.nagUserOfDiscontinuedMethod("TaskInputs.getProperties()", "Use the property() and properties() methods to declare input properties instead.");
+        return getAllProperties();
+    }
+
+    @Override
+    public Map<String, Object> getAllProperties() {
```

Comment:
I don't like having a new method which still exposes all the task properties. This method still bleeds into the public API via `TaskPropertiesInternal`. Then we could/should leave the old query methods in, too.
If it is only for tests, we could leave the method in `DefaultTaskInputs` and not add it to any interface. We would still be able to call it from Groovy.

Maybe it would be better to have a test utility method which would return the task input properties. WDYT?

Reply:
Good point. I fixed it.